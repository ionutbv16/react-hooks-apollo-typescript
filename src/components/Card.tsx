import React from 'react';
import styled from 'styled-components';
import Card from "@material-ui/core/Card";
import { Link } from 'react-router-dom';
import { CardProps } from '../types';


const Wrapper = styled.div`
  font-size: 14px;
  flex:2;
  padding: 14px;
  display: flex;
  align-items: center;
  :hover {
    box-shadow: 0px 1px 2px 0px #d3d3d3;
  }
`;

const CardRow = styled.div`
  flex: 2;
  padding: 15px;
`;

const CardWinner = styled.div`
  flex: 2;
  padding: 15px;
  font-weight: bold;
  color: green
`;

const CardLoser = styled.div`
  flex: 2;
  padding: 15px;
  font-weight: bold;
`;

const LinkWrapper = styled.div`
  color: #006699;
  text-decoration:none;
  font-weight: bold;
`;


const CardComp: React.FC<CardProps> = ({ id, name, hyperdriverating, height, type, winner }) => (
  <Wrapper>
    <Card>
      {type && <CardRow> {name}</CardRow>}
      { (type === 'People') &&  <CardRow>Height: {height}</CardRow>} 
      { (type === 'Starships') &&  <CardRow>Hyperdrive Rating: {hyperdriverating}</CardRow>} 
      { type && ( winner ? <CardWinner>Winner</CardWinner> : <CardLoser>Loser</CardLoser> ) } 
      { !type  && <CardRow><Link to={`/game/${name}`} ><LinkWrapper>{name}</LinkWrapper></Link></CardRow> }
    </Card>
  </Wrapper>
);

export default CardComp;
