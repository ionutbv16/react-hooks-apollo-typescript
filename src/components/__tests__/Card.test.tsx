import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router';
import Card from '../Card';

describe('Test Card Component', () => {
    it('should render Loading without error', () => {
    const props = {
        name: 'People'
    }
    const component = renderer.create(<MemoryRouter><Card {...props} /></MemoryRouter>);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
    });
});
