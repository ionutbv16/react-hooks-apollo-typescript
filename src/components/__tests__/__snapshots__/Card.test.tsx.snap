// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Card Component should render Loading without error 1`] = `
<div
  className="sc-bdVaJa kgqLdc"
>
  <div
    className="MuiPaper-root MuiPaper-elevation1 MuiCard-root MuiPaper-rounded"
  >
    <div
      className="sc-bwzfXH fdVHAP"
    >
      <a
        href="/game/People"
        onClick={[Function]}
      >
        <div
          className="sc-ifAKCX bZbatV"
        >
          People
        </div>
      </a>
    </div>
  </div>
</div>
`;

exports[`should render Loading without error 1`] = `
<div
  className="sc-bdVaJa kgqLdc"
>
  <div
    className="MuiPaper-root MuiPaper-elevation1 MuiCard-root MuiPaper-rounded"
  >
    <div
      className="sc-bwzfXH fdVHAP"
    >
      <a
        href="/game/People"
        onClick={[Function]}
      >
        <div
          className="sc-ifAKCX bZbatV"
        >
          People
        </div>
      </a>
    </div>
  </div>
</div>
`;
